app/src/main/java/com/nextcloud/talk/chat/ChatActivity.kt

```diff
@@ -2600,11 +2600,10 @@ class ChatActivity :
             chatMessage.activeUser = conversationUser
 
             adapter?.let {
-                chatMessage.isGrouped = (
-                    it.isPreviousSameAuthor(chatMessage.actorId, -1) &&
-                        it.getSameAuthorLastMessagesCount(chatMessage.actorId) %
-                        GROUPED_MESSAGES_SAME_AUTHOR_THRESHOLD > 0
-                    )
+                val lastItemIndex = adapter!!.itemCount - 1
+                val previousChatMessage = adapter!!.items[lastItemIndex].item as ChatMessage
```

Comment:
`previousChatMessage` is not calculated correctly.

When querying
`adapter!!.itemCount - 1`

you would get some old messages (the oldest one available in the adapter).

The newest message has index 0

So it's just
`val previousChatMessage = adapter!!.items[0].item as ChatMessage`

But also it must be checked if the message is a ChatMessage. Otherwise there will be cast exceptions, for example when it's a date-message (the ones that are shown between the messages, saying "Today", "Yesterday", ...). Just look into the implementation of `isPreviousSameAuthor` to see the checks that may be necessary.